healthcheck/handler.go

```diff
@@ -0,0 +1,49 @@
+package healthcheck
+
+import (
+	"net/http"
+	"github.com/ONSdigital/go-ns/log"
+	"github.com/ONSdigital/go-ns/server"
+	"github.com/gorilla/mux"
+	"sync"
+	"context"
+	"time"
+)
+
+
+const (
+	gracefulShutdownMsg = "Graceful shutdown of healthcheck endpoint complete"
+)
+
+var httpServer *server.Server
+var once sync.Once
+
+// NewHandler create and run the healthcheck API endpoint.
+func NewHandler(bindAddr string, shutdownTimeout time.Duration) {
+	once.Do(func() {
+		router := mux.NewRouter()
+		router.Path("/healthcheck").HandlerFunc(handle)
+
+		httpServer = server.New(bindAddr, router)
+		// Disable this here to allow main to manage graceful shutdown of the entire app.
+		httpServer.HandleOSSignals = false
+		httpServer.DefaultShutdownTimeout = shutdownTimeout
+
+		go func() {
+			log.Debug("Starting healthcheck endpoint...", nil)
+			if err := httpServer.ListenAndServe(); err != nil {
+				log.ErrorC("healthcheck server returned error", err, nil)
```

Comment:
Consider using the 'error channel' here to pass an error back up to main for a graceful shutdown. If the http server has issues then the healthchecks will fail, so we may as well shutdown the app gracefully.